File: cpp/q002.cpp
#include <iostream>
#include <string>
using namespace std;

int fibo(int n){
    if (n==1 or n==2){
        return n;
    }
    return fibo(n-1)+fibo(n-2);
}

int main(){
    int i=1;
    int ans=0;
    while (fibo(i)<4000000){
        ans += fibo(i);
        i++;
    }
    cout << ans << endl;
    return 0;
}

File: rust/q002.rs
fn fibo(x: i32) -> i32 {
    if x == 1 || x == 2 {
        x
    } else {
        fibo(x - 1) + fibo(x - 2)
    }
}

fn main() {
    let mut x: i32 = 1;
    let mut ans: i32 = 0;
    let mut f: i32 = 0;
    while f <= 4000000 {
        f = fibo(x);
        x += 1;
        if f % 2 == 0 {
            ans += f;
        }
    }

    println!("{}", ans)
}
